### ISSUE TITLE
Port label inside port?

### ISSUE THREAD

**Issue by diegosogari** [eclipse/elk#970]
Hello. I've taken interest in developing a new diagram type for [mermaidjs](https://mermaid.js.org/) using [elkjs](https://github.com/kieler/elkjs). It is based on the [hexagonal architectural pattern](https://www.wikiwand.com/en/Hexagonal_architecture_(software)).

In this kind of diagram, software components are modeled as concentric layers (usually of hexagonal shape). Each layer may have a set of interfaces for interacting with its immediate parent layer. I can use ELK's nodes to represent layers and their elements, but I want to use ELK's port concept to represent the interfaces.

Currently, the [Port Label Placement](https://eclipse.dev/elk/reference/options/org-eclipse-elk-portLabels-placement.html) layout option does not provide a way to place port labels inside the ports themselves. Hence, the workaround would be to set the port size according to the size that the label would have, and place the text inside that rectangle while rendering the graph.

Am I correct in my assumptions? Would it be possible to add an option for placing port labels inside the ports?

**Comment by soerendomroes** [comment]
This is correct. `portLabel.placement: INSIDE` is set on nodes and moves the labels inside the node (not the port), as seen [here](https://rtsys.informatik.uni-kiel.de/elklive/elkgraph.html?compressedContent=IYGw5g9gTglgLgCwLYC4AEJgE8CmUcAmAUEQHYQE5qkCMaA3kWs9RTgMowBeOAdAMYRSAZzhRgMUnGHoARAAUA8gCUAKuzRK1AfQAyAQQBCAUV3tZTFgAdocXcABGOEMN5XM-HEhxT0ASQA5dj8AEWNLZhsoODQrOkYWRIxHZzRZVRxRYGyc2TQAeny0VQQqYStgTzQYYTQCCABXBxBCNCd+YAbhKggAMzRhLCRvMSw0AHcYfgQ0DtI2qgIaxxaCXgi0AF8ibfJKagAmMjZqAGYiQjAqWjQAWgA+Q4uCK+o6B7OgA).

Your workaround sounds reasonable. Generally, ELK cannot calculate the size of a label but only places it based on a precomputed size. Hence, you should also know the size the port should have and must only calculate the position of the label.

**Comment by diegosogari** [comment]
@soerendomroes Thanks! That's what I thought.
